xml2rfc_compat/serializers/reference.py

```diff
@@ -106,19 +107,30 @@ def create_reference(item: BibliographicItem) -> _Element:
             func(docid)
             for func in DOCID_SERIES_EXTRACTORS
         ])
+
+    is_doi = False
     for series_info in list(dict.fromkeys(series)):
         if series_info is not None:
             ref.append(E.seriesInfo(
                 name=series_info[0],
                 value=series_info[1],
             ))
+            if series_info[0] == "DOI":
+                is_doi = True
 
     # Target, may be overwritten by callers
     try:
         target = get_suitable_target(as_list(item.link or []))
     except ValueError:
         pass
     else:
+        if is_doi:
+            # https://github.com/ietf-tools/bibxml-service/issues/332
+            url_parse = urlparse(target)
+            if url_parse.hostname == "dx.doi.org":
+                target = target.replace("//dx.doi.org", "//doi.org")
```

Comment:
Better to use `._replace()` method for `netloc` and `scheme` (line 133) instead of `str.replace()`?
See https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse examples.